compliance/cis_k8s/rules/cis_1_2_4/data.yaml

```diff
@@ -1,31 +1,41 @@
 metadata:
-  id: d6fc4ba2-5cdf-5ef2-be01-451ec7ded8bc
-  name: Ensure that the --kubelet-https argument is set to true (Automated)
+  id: e0b17dee-7c45-51b0-9000-ab4c821cbfcc
+  name: Ensure that the --kubelet-client-certificate and
```

Comment:
many of the names of the rules strings are cut midway, is this intentional?

Reply:
We are parsing according to how they are being written in the PDF.
If the rule is being written as a two-liner in the PDF then you will see these cuts.
It is only reflected in the YAML, but when you are requiring within code or converting to JSON it is irrelevant.
```json
{
  "metadata": {
    "id": "e0b17dee-7c45-51b0-9000-ab4c821cbfcc",
    "name": "Ensure that the --kubelet-client-certificate and --kubelet-client-keyarguments are set as appropriate (Automated)",
    "profile_applicability": "* Level 1 - Master Node\n",
    "description": "Enable certificate based kubelet authentication.\n",
    "rationale": "The apiserver, by default, does not authenticate itself to the kubelet's HTTPS endpoints. The requests from the apiserver are treated anonymously. You should set up certificate- based kubelet authentication to ensure that the apiserver authenticates itself to kubelets when submitting requests.\n",
    "audit": "Run the following command on the master node:\n```\nps -ef | grep kube-apiserver\n```\nVerify that the `--kubelet-client-certificate` and `--kubelet-client-key` arguments\nexist and they are set as appropriate.\n",
    "remediation": "Follow the Kubernetes documentation and set up the TLS connection between\nthe\napiserver and kubelets. Then, edit API server pod specification file\n`/etc/kubernetes/manifests/kube-apiserver.yaml` on the master node and set the\nkubelet client certificate and key parameters as below.\n```\n--kubelet-client-certificate=<path/to/client-certificate-file>\n--kubelet-client-key=<path/to/client-key-file>\n```\n",
    "impact": "You require TLS to be configured on apiserver as well as kubelets.\n",
    "default_value": "By default, certificate-based kubelet authentication is not set.\n",
    "references": "1. [https://kubernetes.io/docs/admin/kube-apiserver/](https://kubernetes.io/docs/admin/kube-apiserver/)\n2. [https://kubernetes.io/docs/admin/kubelet-authentication-authorization/](https://kubernetes.io/docs/admin/kubelet-authentication-authorization/)\n3. [https://kubernetes.io/docs/concepts/cluster-administration/master-node-communication/#apiserver---kubelet](https://kubernetes.io/docs/concepts/cluster-administration/master-node-communication/#apiserver---kubelet)\n",
    "section": "API Server",
    "version": "1.0",
    "tags": [
      "CIS",
      "Kubernetes",
      "CIS 1.2.4",
      "API Server"
    ],
    "benchmark": {
      "name": "CIS Kubernetes V1.20",
      "version": "v1.0.0"
    }
  }
}
```
Notice that the name is being parsed as a single row string.

Reply:
I think there is a bigger Issue here @ofiriro3 turned my attention to, and these are the newlines in the middle of other fields. they break the lines where the pdf does and this is not always right as it might cause weird look:
<img width="548" alt="Screen Shot 2022-04-20 at 14 57 37" src="https://user-images.githubusercontent.com/63912106/164225358-2eaf3f26-bceb-4369-ad80-a3b44abc6cb3.png">

also another thing I understood later is that markdown only dates into consideration double newline or newlines before special elements like codeblocks. which makes most of the newlines in the current yamls useless.
Also I'm guessing the script breaks apart `inline code` with newline which is also problematic.
<img width="1094" alt="Screen Shot 2022-04-20 at 15 02 35" src="https://user-images.githubusercontent.com/63912106/164226196-f8708736-e8b6-4012-a735-fb037a2bde76.png">

I think we need to either open an issue to deal with all these problems or reopen https://github.com/elastic/security-team/issues/3380

Reply:
nice finding @eyalkraft. I think this should have its own issue.

Reply:
@eyalkraft 
As shown here [rule to EuiMarkdownFormat example](https://codesandbox.io/s/euimarkdownformat-hyf1kn?file=/data.json), there isn't a problem with single `\n` to mark a new line for the markdown.
Regarding the line breaks, I've opened the issue https://github.com/elastic/security-team/issues/3672.